adcpipeline/pipeline.py

```diff
@@ -1,135 +1,170 @@
+            when the pipeline is run.
+        """
+        return self._method_list
+
+    @classmethod
+    def from_yaml_file(cls, df: DataFrame, path: str):
+        """
+        This is a factory method to instantiate this class by loading the settings from a yaml file.
+        Format of yaml file should be:
+        pipeline:
+          - <method_name>: {<argument_name>: <argument_value>}
+          - <method_name>: {<argument_name>: <argument_value>}
+          - ...
+        Args:
+            df: This is your data in a DataFrame format.
+            path: Path to yaml file.
+
+        Returns:
+            Instance of this class.
+        """
+        settings = LoadConfig.load_yaml_as_dict(path)['pipeline']
+        return cls(df=df, method_settings=settings)
+
+    def __call__(self) -> None:
+        self.run()
+
+    def __repr__(self) -> str:
+        return str(self.method_settings)
+
+    def __getitem__(self, i: int) -> Dict:
+        return self.method_settings[i]
+
+    def __setitem__(self, i: int, setting: Dict) -> None:
+        self.method_settings[i] = setting
+        self._method_list[i] = self.__get_lambda_method(setting)
+
+    def __delitem__(self, i: int) -> None:
+        del self.method_settings[i]
+        del self.method_list[i]
+
+    def __len__(self) -> int:
+        return len(self.method_settings)
+
+    def __iter__(self) -> Iterator[Dict]:
+        return iter(self.method_settings)
+
+    def __reversed__(self) -> List[Dict]:
+        return list(reversed(self.method_settings))
+
+    def insert(self, index: int, setting: Dict) -> None:
+        self.method_settings.insert(index, setting)
+        self.method_list.insert(index, self.__get_lambda_method(setting))
+
+    def reset(self) -> None:
+        """
+        Reset the current step of the pipeline to 0
+        """
+        self._step = 0
+
+    def run(self, steps: int = None) -> None:
+        """
+        This will call all the lambdas (in order) saved in the 'method_list'. These methods can be set with the
+        property 'method_settings'.
+        Args:
+            steps: instead of running all methods in the method list, run an N amount of methods.
+        """
+        if steps is None:
+            steps = len(self.method_list)
+        if self._step >= len(self.method_list):
+            logger.info("Pipeline already ran all methods, run pipeline.reset() before running again")
+            return
+        till_step = min(self._step + steps, len(self.method_settings))
+        current_methods = self.method_settings[self._step: till_step]
+        logger.info(f'Running pipeline using the following settings: {current_methods}')
+        for method in self.method_list[self._step: till_step]:
+            logger.info(f"Executing method {method}")
+            method()
+        self._step += steps
+        if self.path is not None:
+            self.df.to_pickle(self.path)
+
+    def run_or_load(self) -> None:

```

Comment:
Ik zou hier nog onderscheid maken in de naam t.o.v. de reguliere `run` method, aangezien deze method nooit steps zal gebruiken en altijd een full run zal zijn. Misschien is een betere naam iets van `full_run_with_cache` of `full_run_or_load`?